### PR TITLE
gpg-agent: add `enableExtraSocket` and `verbose` options.

### PR DESCRIPTION
**Pull request by gleber**
This option enables a GPG Agent restricted socket (aka "extra-socket"), which
can be used to forward GPG Agent over SSH.

Additionally `verbose` option enables verbose output of an `gpg-agent.service`
unit for easier debugging.

See: https://wiki.gnupg.org/AgentForwarding

### CODE REVIEW

File: modules/services/gpg-agent.nix [hunk 1]
```diff
@@ -115,7 +132,7 @@ in
         };
 
         Service = {
-          ExecStart = "${pkgs.gnupg}/bin/gpg-agent --supervised";
+          ExecStart = "${pkgs.gnupg}/bin/gpg-agent ${if cfg.verbose then "--verbose" else ""} --supervised";
```
Comment (rycee) [inline modules/services/gpg-agent.nix]: I'd suggest using `optionalString`. Something like

```
          ExecStart =
            "${pkgs.gnupg}/bin/gpg-agent --supervised"
            + optionalString cfg.verbose " --verbose";
```

**Comment by rycee** [comment]
Looks good! I just had a minor suggestion about the verbose command option.

File: modules/services/gpg-agent.nix (hunk 1 above)
Comment (gleber) [reply modules/services/gpg-agent.nix]: Done

**Comment by gleber** [comment]
PTAL

**Comment by rycee** [comment]
Rebased to master in 9bf9e7ac5c60ba442f4f0dbc3fea27c09a999ab3. Thanks for the nice contribution!

**Pull request closed without merging**